### ISSUE TITLE
minloc/maxloc reductions over associative arrays with idxType string

### ISSUE THREAD

**Issue by ben-albrecht** [chapel-lang/chapel#6311]
### Summary of Problem

Calling `minloc` / `maxloc` reductions over associative arrays declared over `domain(string)` results in an error.

The code shown below results in the following error:

```
$CHPL_HOME/modules/internal/ChapelTuple.chpl:364: In function 'min':
$CHPL_HOME/modules/internal/ChapelTuple.chpl:367: error: unresolved call 'min(type string)'
$CHPL_HOME/modules/internal/ChapelTuple.chpl:367: note:   instantiated from min
$CHPL_HOME/modules/internal/ChapelReduce.chpl:256: note:   instantiated from _type_construct_maxloc
...
```

The problem is caused by the maxloc reduction trying to call `min(string)` to get an identity value for the index type. From what I understand, the `maxloc` reduction should not need to call `min()` on the index type. I suspect this would occur for any index type that is not supported by `min()`, but I haven't test extensively.

### Steps to Reproduce

**Source Code:**

```chapel
var D: domain(string);
var A : [D] int;

A['a'] = 1;
A['b'] = 2;
A['c'] = 3;

const (maxValue, i) = maxloc reduce zip(A, D);
```

**Associated Future Test(s):**

*TODO*

**Comment by benharsh** [comment]
I think it calls ``min`` because it needs to set some kind of default value for the index in case multiple indices map to the maximum value. I believe it chooses the lower of the two indices in such cases.

It should probably be ``(min(eltType), max(idxType))``, but that doesn't really help strings.

**Comment by benharsh** [comment]
Should this be closed in favor of #6313 ?

**Comment by ben-albrecht** [comment]
Sure thing!
